From bb1eaf67be65c847c1d52b04b0a74c74ac47d8e6 Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Sat, 6 May 2023 15:20:07 -0700
Subject: [PATCH 1/6] imx8mn_var_som: runtime image offset detection for imx8mn

Add support of next image ram offset detection (in SPL) for
IMX8MN_VAR_SOM.

Signed-off-by: Michael Scott <mike@foundries.io>
---
 include/configs/imx8mn_var_som.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/configs/imx8mn_var_som.h b/include/configs/imx8mn_var_som.h
index 69b10293a41..d609bdafa22 100644
--- a/include/configs/imx8mn_var_som.h
+++ b/include/configs/imx8mn_var_som.h
@@ -20,6 +20,16 @@
 #define CONFIG_SYS_BOOTM_LEN		(32 * SZ_1M)
 #define CONFIG_SPL_MAX_SIZE		(208 * 1024)
 #define CONFIG_SYS_MONITOR_LEN		(512 * 1024)
+/*
+ * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET was intended to be used as a compile time setting.
+ * If CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION is defined, then only use the base offset of 0x300 here.
+ * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET will be added dynamically if needed.
+ */
+#ifndef CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION
+#undef CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR
+#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR        (0x300 + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET)
+#endif
+
 #define CONFIG_SYS_MMCSD_FS_BOOT_PARTITION	1
 #define CONFIG_SYS_UBOOT_BASE		(QSPI0_AMBA_BASE + CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR * 512)
 
-- 
2.40.1

